### PR TITLE
Bokeh plotting 2->3 fix second (final?) round

### PR DESCRIPTION
**Pull request by EricThomson**
# Description

As far as I can tell, fixes the residual issues brought up in #1127 and previous PR #1129 . The problem was that images were inverted in ways I didn't notice, and this was percolating into almost every visualization function. I think it is now fixed in all tools (2d and 3d). 

# Type of change
[X] Bug fix (non-breaking change which fixes an issue)

# Has your PR been tested?
Yes against test and demotest, and I also tested with the following pipelines:

    demo_pipeline
    demo_pipeline_CNMFE
    demo_caiman_cnmf_3D
    demo_online_cnmfE
    demo_online_3d

These all seemed to work well!

@j-friedrich it would be helpful if you tested that this is indeed reasonable (esp in `demo_caiman_cnmf_3D.ipynb` which is where thigns broke before: I think `axis=2` is working well now too for the 3d component viewer with max projection, which is really nice, as that is the more intuitive projection).

### CODE REVIEW

File: caiman/source_extraction/cnmf/estimates.py [hunk 1]
```diff
@@ -249,7 +249,6 @@ def plot_contours_nb(self, img=None, idx=None, thr_method='max',
                 set of dictionary containing the various parameters
         """
         try:
-            import bokeh
```
Comment (pgunn) [inline caiman/source_extraction/cnmf/estimates.py:252]: Awesome to see imports moved out of conditionals

File: demos/notebooks/demo_caiman_cnmf_3D.ipynb [hunk 2]
```diff
@@ -546,10 +543,7 @@
    "metadata": {},
    "source": [
     "### View the results\n",
-    "For a change we view the components as max-projections (frontal in the XY direction, sagittal in YZ direction and transverse in XZ), and we additionaly show the denoised trace\n",
-    "\n",
-    "**Note `nb_view_components_3d()` is buggy since a new Bokeh release, and we will work on a fix soon.     \n",
-    "If you know Bokeh and would like to help, please let us know!**"
+    "For a change we view the components as max-projections (frontal in the XY direction, sagittal in YZ direction and transverse in XZ), and we additionaly show the denoised trace"
```
Comment (pgunn) [inline demos/notebooks/demo_caiman_cnmf_3D.ipynb:546]: I know it was the original wording, but "For a change" is really weird phrasing; might be worth rephrasing it while it's being touched.

**Comment by pgunn** [comment]
Generally LGTM

**Comment by pgunn** [comment]
oasis.cpp really needs to be removed from this; if you don't mind removing it from the history of this branch that'd be ideal.

**Comment by EricThomson** [comment]
> oasis.cpp really needs to be removed from this; if you don't mind removing it from the history of this branch that'd be ideal.

I keep trying new ways to eradicate that and this time I ended up committing it. I wonder if we add `*oasis.cpp` to `.gitignore` we'll have more success.

**Pull request merged**